tests/sentry/api/endpoints/test_integrations_slack_commands.py

```diff
@@ -53,54 +86,70 @@ def setUp(self):
             },
         )
         self.integration.add_organization(self.organization, self.user)
+        self.idp = IdentityProvider.objects.create(type="slack", external_id="slack:1", config={})
+        self.login_as(self.user)
 
 
 class SlackCommandsGetTest(SlackCommandsTest):
+    method = "get"
+
     def test_method_get_not_allowed(self):
         self.get_error_response(status_code=status.HTTP_405_METHOD_NOT_ALLOWED)
 
 
-class SlackCommandsHelpTest(SlackCommandsTest):
-    method = "post"
-
+class SlackCommandsPostTest(SlackCommandsTest):
     def test_invalid_signature(self):
-        # The `get_error_response` method doesn't use a signature.
+        # The `get_error_response` method doesn't add a signature to the request.
         self.get_error_response(status_code=status.HTTP_400_BAD_REQUEST)
 
     def test_missing_team(self):
-        self.get_slack_response({"text": ""}, status_code=status.HTTP_403_FORBIDDEN)
+        self.get_slack_response({"text": ""}, status_code=status.HTTP_400_BAD_REQUEST)
+
+    def test_idp_does_not_exist(self):
+        """Test that get_identity fails if we cannot find a matching idp."""
+        data = self.send_slack_message("", team_id="slack:2")
+        assert DISCONNECTED_MESSAGE in get_response_text(data)
 
+
+class SlackCommandsHelpTest(SlackCommandsTest):
     def test_missing_command(self):
-        response = self.get_slack_response({"text": "", "team_id": self.external_id})
-        assert_is_help_text(response)
+        data = self.send_slack_message("")
+        assert_is_help_text(data)
 
     def test_invalid_command(self):
-        response = self.get_slack_response({"text": "invalid command", "team_id": self.external_id})
-        assert_is_help_text(response, "invalid")
+        data = self.send_slack_message("invalid command")
+        assert_is_help_text(data, "invalid")
 
     def test_help_command(self):
-        response = self.get_slack_response({"text": "help", "team_id": self.external_id})
-        assert_is_help_text(response)
+        data = self.send_slack_message("help")
+        assert_is_help_text(data)
 
 
-class SlackCommandsLinkTeamTest(SlackCommandsTest):
-    method = "post"
+class SlackCommandsLinkTest(SlackCommandsTest):
```

Comment:
can you write `SlackCommandsLinkUserTest` tests to run through it all and make sure the templates are used and the identity is created / destroyed?

Reply:
I only added a simple test because the auth flow is already being tested by the other slack link.